reporthandling/attacktrack/v1alpha1/attacktrackmethods_test.go

```diff
@@ -294,3 +294,396 @@ func TestNewAttackTrackControlsLookup(t *testing.T) {
 
 	assert.JSONEq(t, string(expectedJson), string(resultJson))
 }
+
+func TestCalculatePathsRootToLeaf(t *testing.T) {
+	tests := []struct {
+		name        string
+		attackTrack IAttackTrack
+		controlsMap AttackTrackControlsLookup
+		want        [][]string
+	}{
+		{
+			name: "Found attack chain",
+			attackTrack: GetAttackTrackMock(
+				AttackTrackStep{
+					Name: "A",
+					SubSteps: []AttackTrackStep{
+						{
+							Name: "C",
+							SubSteps: []AttackTrackStep{
+								{
+									Name: "B",
+								},
+								{
+									Name: "D",
+								},
+								{
+									Name: "E",
+									SubSteps: []AttackTrackStep{
+										{
+											Name: "G",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "F",
+						},
+					},
+				},
+			),
+			controlsMap: AttackTrackControlsLookup{
+				"TestAttackTrack": {
+					"A": {
+						control_1, control_2,
+					},
+					"F": {
+						control_3, control_4,
+					},
+					"C": {
+						control_5,
+					},
+					"B": {
+						control_6,
+					},
+					"G": {
+						control_7, control_8,
+					},
+				},
+			},
+			want: [][]string{
+				{"A", "C", "B"},
+				{"A", "F"},
+			},
+		},
+		{
+			name: "No attack chain",
+			attackTrack: GetAttackTrackMock(
+				AttackTrackStep{
+					Name: "A",
+					SubSteps: []AttackTrackStep{
+						{
+							Name: "C",
+							SubSteps: []AttackTrackStep{
+								{
+									Name: "B",
+								},
+								{
+									Name: "D",
+								},
+								{
+									Name: "E",
+									SubSteps: []AttackTrackStep{
+										{
+											Name: "G",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "F",
+						},
+					},
+				},
+			),
+			controlsMap: AttackTrackControlsLookup{
+				"TestAttackTrack": {
+					"A": {
+						control_1, control_2,
+					},
+					"B": {
+						control_6,
+					},
+				},
+			},
+			want: nil,
```

Comment:
I can see that  `CalculatePathsRootToLeaf` never returns nil - if no attack chains found it will return an empty list but  by reading the test this is what I assume.
I suggest to change it to what we expect the function to return (in this case an empty list) - this way you can also remove the redundant condition in line 411 so it will not be a special case for the test.
The same applies to the other tests (which I see expect `nil`)